p2p/host/basic/basic_host.go

```diff
@@ -131,7 +131,7 @@ type HostOpts struct {
 
 	// NATManager takes care of setting NAT port mappings, and discovering external addresses.
 	// If omitted, this will simply be disabled.
-	NATManager func(network.Network) NATManager
+	NATManager func(network.Network, Option) (NATManager, error)
```

Comment:
The beauty of the option pattern is that the options are optional. You need to make them variadic though.

```suggestion
	NATManager func(network.Network, ...Option) (NATManager, error)
```

Reply:
@marten-seemann Ready. But I still think it's redundant to add an Option for a single parameter. Perhaps there will be more parameters in the future?

Reply:
Fixed the error